### DISCUSSION TITLE
Attempting to bridge the EMQX with Ali IOT platform

### DISCUSSION THREAD

**Discussion by LZY-SPCA** [emqx/emqx#12409] [Q&A]
I'm trying to bridge EMQX with Aliyun IOT platform but I couldn't edit clientID when creating a bridge through MQTT so I cannot connect to the platform. How can I edit clientID when creating a connector(or bridge)?

**Comment by SergeTupchiy** [comment]
Hi @LZY-SPCA,

It's not possible to set clientid in MQTT bridge config, but:
- bridge name is used as a base for a clientid, so you can set a bridge name to a desired clientid.
  For example, if bridge name is "mybridge", the clientid would look something like: "test2bd61c44:1", where 2bd61c44 is a hash of EMQX node name, and 1 is a id of a connection (the bridge creates a pool of several connections (configurable in bridge settings). 
  Node hash and connection id are added to avoid clientid clashes.
- `clientid_prefix` can be added to bridge config (no UI for that, needs to be set in cluster.hocon), for example:
```
bridges {
  mqtt {
    test {
      clientid_prefix = myprefix
...
```
Then the clientid would look like: `myprefix:test2bd61c44:1`
Please note that the length of clientid in any case is limited to 23 bytes and if it is exceeded it will be replaced with a 23 bytes hash value.
